### ISSUE TITLE
Syllables are cached incorrectly 

### ISSUE THREAD

**Issue by StylishTriangles** [andreasvc/readability#7]
Hi,
I have run into a rare issue with syllable counting. Due to the way that syllables are currently cached, running readability's getmeasures on the same string twice can produce different results.
Consider the following string: "per se, see" what happens on the first run is this:
1. check for se in cache - no results
2. _se_ gets it's last _e_ removed, _s_ remains
3. there are 0 syllables in _s_ so cache['s'] = 0
4. check for see in cache - no results
5. _see_ gets its last _e_ removed, _se_ remains
6. there is 1 syllable in se, so cache['se'] = 1

Now on the second run we will be checking for _se_ from phrase _per se_ in cache. And we will find 1! which will cause "per se, see" to have 3 syllables on the second run.

I propose #6 to fix this issue.

**Comment by andreasvc** [comment]
Good catch. I applied your pull request. It's clear that the syllable counting is very much a heuristic approximation, but now at least it should give consistent results.
